lib/entities/team-space-membership.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Space Membership instances
+ * @namespace SpaceMembership
+ */
+
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+import {
+  createDeleteEntity
+} from '../instance-actions'
+import errorHandler from '../error-handler'
+
+/**
+ * @memberof TeamSpaceMembership
+ * @typedef TeamSpaceMembership
+ * @prop {Meta.Sys} sys - System metadata
+ * @prop {boolean} admin - User is an admin
+ * @prop {array} roles - Array of Role Links
+ * @prop {function(): Object} toPlainObject() - Returns this Team Space Membership as a plain JS object
+ */
+
+function createTeamSpaceMembershipApi (http) {
+  return {
+
+    /**
+     * Sends an update to the server with any changes made to the object's properties
+     * @memberof TeamSpaceMembership
+     * @func update
+     * @return {Promise<SpaceMembership>} Object returned from the server with updated changes.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getSpace('<space_id>')
+     *  .then((space) => space.getTeamSpaceMembership('team_space_membership_id'))
+     *  .then((teamSpaceMembership) => {
+     *    item.roles = [
+     *      {
+     *        sys: {
+     *          type: 'Link',
+     *          linkType: 'Role',
+     *          id: 'role_id'
+     *        }
+     *      }
+     *    ]
+     *  })
+     *  .then((spaceMembership) => console.log(`spaceMembership ${spaceMembership.sys.id} updated.`))
+     *  .catch(console.error)
+     */
+    update: function () {
+      const raw = this.toPlainObject()
+      const data = cloneDeep(raw)
```

Comment:
i am not sure if it is required or not,  but last endpoint I did I had to add `delete data.sys` (after cloneDeep) to make the update endpoint work since we do not allow to update sys data I think

Reply:
@vbeleuta yeah, I think so too. fixed.